src/Core/src/Core/IStartup.cs

```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.Maui.Hosting;
+
+namespace Microsoft.Maui
+{
+	/// <summary>
+	/// Startup allow to configure and instantiate application services.
+	/// </summary>
+	public interface IStartup
+	{
+		/// <summary>
+		/// Configures the application.
+		/// Configure are called by the .NET MAUI Core runtime when the app starts.
+		/// </summary>
+		/// <param name="appBuilder">Defines a class that provides the mechanisms to configure an application's dependencies.</param>
+		void Configure(IAppHostBuilder appBuilder);
```

Comment:
@rmarinho What do you think? Maybe we should create directly the Builder? `virtual IAppHostBuilder CreateBuilder()`

Reply:
We might want both? I think we want to configure it still but have a:

```
public virtual IAppHostBuilder CreateBuilder() => new DefaultAppHostBuilder();
```

So we can choose to override it, or let the default implementation be created.

Reply:
yeah agree with @Redth , we want it virtual that calls the default builder, user can override

Reply:
I would add a new Method that is CreateBuilder that returns CreateDefaultBuilder.

Reply:
In ASP.NET, the user instantiates the builder in Program.cs and then configures it in Startup.cs. Now this is not quite possible here, but we can have our Startup have a "get builder" method instead of configure? Or maybe we should have an additional `IHostBuilderStartup` interface that will allow the user top optionally implement it and construct the builder?

What if the user does not want our defaults? For example, not to register all the MAUI views and not to use the fonts. And, this might have future implications. Just as we had launch options in Forms to control what was registered, we may need this so users can not have to pass in some options, but rather control the whole thing,